diff -ur old/includes/EditPage.php new/includes/EditPage.php
--- includes/EditPage.php	2008-10-02 11:22:34.000000000 -0500
+++ includes/EditPage.php.new	2008-10-02 11:31:08.000000000 -0500
@@ -1559,6 +1559,9 @@
 			$wgOut->addHTML( $parserOutput->mText );
 			$previewHTML = '';
 		} else {
+			# Preview protection by Jon Eisenstein for improved-accesscontrol extension
+			wfRunHooks( 'EditFilter', array( $this, $this->textbox1, $this->section, &$this->hookError, $this->summary ) );
+			
 			$toparse = $this->textbox1;
 
 			# If we're adding a comment, we need to show the
diff -ur old/includes/specials/SpecialSearch.php new/includes/specials/SpecialSearch.php
--- includes/specials/SpecialSearch.php	2008-10-02 11:22:54.000000000 -0500
+++ includes/specials/SpecialSearch.php.new	2008-10-02 11:32:50.000000000 -0500
@@ -376,7 +376,14 @@
 		//Note that hiding the entry entirely would screw up paging.
 		if (!$t->userCanRead()) {
 			wfProfileOut( $fname );
-			return "<li>{$link}</li>\n";
+			
+			# Added by Jon Eisenstein for improved-accesscontrol extension
+			global $wgHideProtectedSearchResultTitles;
+			if (!$wgHideProtectedSearchResultTitles) {
+				return "<li>{$link}</li>\n";
+			} else {
+				return "";
+			}
 		}
 
 		// If the page doesn't *exist*... our search index is out of date.
